test/singer_clojure/catalog_test.clj

```diff
@@ -0,0 +1,93 @@
+(ns singer-clojure.catalog-test
```

Comment:
I think this looks familiar, just wanted to comment that there should be a bunch of tests for this in tap-mssql, if you didn't copy this from there, that might be a gold mine of conditions.

Reply:
Yeah we'll take a look. Obviously we took code from the existing taps and sometimes tests too. We'll take one more look there in case.

Reply:
So all those tests are creating and destroying databases, which is specific to ms sql and not necessarily the library code. We looked through them to see if there was an edge case we missed but we believe the test we have is end-to-end enough to cover what we want to have tested.

Reply:
We were checking that out and most of them weren't generic enough to put into the library, so we decided to just write a super generic serialize/deserialize test

Reply:
Cool, sounds good